### PR TITLE
v3.0 go

### PR DESCRIPTION
**Pull request by ksyeo1010**

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -55,12 +55,18 @@ jobs:
 
     strategy:
       matrix:
-        os: [ubuntu-latest, macos-latest, windows-latest]
-        go: [ '1.16', '1.17', '1.18' ]
+        os: [ubuntu-latest, macos-latest, windows-latest]
```
Comment (laves) [inline .github/workflows/go.yml:58]: not a big deal, but you should tell you editor to trim spaces on save

File: binding/go/rhino.go [hunk 2]
```diff
@@ -57,12 +57,23 @@ const (
...
+	var message strings.Builder
+	message.WriteString(fmt.Sprintf("%s: %s", pvStatusToString(e.StatusCode), e.Message))
+
+	if len(e.MessageStack) > 0 {
+		message.WriteString(":")
+	}
+
+	for i, value := range e.MessageStack {
+		message.WriteString(fmt.Sprintf("\n  [%d] %s", i, value))
+	}
```
Comment (laves) [inline binding/go/rhino.go:75]: don't we usually put this in the `if` statement? I know it doesn't change the behaviour, but it does read better

File: binding/go/rhino.go [hunk 3]
```diff
@@ -206,39 +217,48 @@ func (rhino *Rhino) Init() error {
...
+				Message:      "Unable to get Rhino error state",
+			}
+		}
+
 		return &RhinoError{
-			PvStatus(status),
-			"Rhino init failed"}
+			StatusCode:		PvStatus(status),
+			Message:		"Rhino init failed",
+			MessageStack: 	messageStack}
```
Comment (laves) [inline binding/go/rhino.go:261]: is all this weird spacing Go's formatter? If so, it's fine, I'm just surprised how random it looks at time

File: binding/go/rhino_test.go [hunk 4]
```diff
@@ -20,7 +20,7 @@ import (
 	"math"
 	"os"
 	"path/filepath"
-	"reflect"
+	"reflect"
```
Comment (laves) [inline binding/go/rhino_test.go:23]: format

**Review by laves — CHANGES_REQUESTED** [review]

**Review by laves — COMMENTED** [review]
there was more than just that one format issue in `rhino.go` - did you run a formatter on the file or just fix the one I pointed out?

**Comment by ksyeo1010** [comment]
I ran it but I guess doesnt convert the go tabs into spaces

**Review by laves — APPROVED** [review]

**Pull request merged**